### PR TITLE
Split up the overwrite test into multiple, more specific tests

### PR DESCRIPTION
**Pull request by passaro**
## Description of change

Split up the overwrite test into multiple, more specific tests.

## Does this change impact existing behavior?

N/A

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

File: mountpoint-s3/tests/fuse_tests/write_test.rs [hunk 1]
```diff
@@ -732,41 +766,151 @@ where
...
+    overwrite_no_truncate_test(fuse::s3_session::new, "overwrite_no_truncate_test", write_only);
+}
+
+#[test_case(true; "write_only")]
+#[test_case(false; "read_write")]
+fn overwrite_no_truncate_test_mock(write_only: bool) {
+    overwrite_no_truncate_test(fuse::mock_session::new, "overwrite_no_truncate_test", write_only);
+}
+
+fn overwrite_truncate_test<F>(creator_fn: F, prefix: &str, write_only: bool)
```
Comment (dannycjones) [inline mountpoint-s3/tests/fuse_tests/write_test.rs:888]: Does this test still have the race condition of file handle being opened for reading too soon after a write, resulting in bad FD?

File: mountpoint-s3/tests/fuse_tests/write_test.rs [hunk 2]
```diff
@@ -778,21 +922,47 @@ where
...
+}
 
-    // Check the new file content
+#[test_case(true; "write_only")]
+#[test_case(false; "read_write")]
+fn overwrite_truncate_test_mock(write_only: bool) {
+    overwrite_truncate_test(fuse::mock_session::new, "overwrite_truncate_test", write_only);
+}
+
+fn read_overwrite_read_test<F>(creator_fn: F, prefix: &str)
```
Comment (dannycjones) [inline mountpoint-s3/tests/fuse_tests/write_test.rs:940]: Similarly, do we expect to see bad FD if the read occurs too soon after the write or write too soon after read?

Comment (dannycjones) [reply mountpoint-s3/tests/fuse_tests/write_test.rs:940]: This is currently passing on `main`, we'll review again when changing behavior.

File: mountpoint-s3/tests/fuse_tests/write_test.rs (hunk 1 above)
Comment (dannycjones) [reply mountpoint-s3/tests/fuse_tests/write_test.rs:888]: This is currently passing on `main`, we'll review again when changing behavior.

**Review by dannycjones — APPROVED** [review]

File: mountpoint-s3/tests/fuse_tests/write_test.rs (hunk 1 above)
Comment (passaro) [reply mountpoint-s3/tests/fuse_tests/write_test.rs:888]: Yes, you are right. But nothing has changed in this PR.

File: mountpoint-s3/tests/fuse_tests/write_test.rs (hunk 2 above)
Comment (passaro) [reply mountpoint-s3/tests/fuse_tests/write_test.rs:940]: read after write, yes. write after read is safe on `main` (may not be if we merge #751)

**Pull request merged**